bpo-40484: Document compiler flags under AST module (#19885)
python/cpython: Doc/library/ast.rst

      Added the *indent* option.

.. _ast-compiler-flags:

> Compiler Flags

Reviewer: I understand why you documented these here, but I am worried about the discovery of these flags. Should it be a cross-reference in the `compile` function docs? Are these even mentioned there?

Author: IMHO this is the initial step to increase that `non-existent` discoverability. I believe we may need to re-write whole flags (and dont_inherit) documentation again for describing them supporting not only future features, but compiler flags in an another PR.

Reviewer: Giving that this PR is rather small, I would suggest doing this in this PR as the decisions we take there may affect this work (for instance we may decide that these flags will be documented only on the `compile` function, rendering this work obsolete.

Author: I made the required adjustments.

--------------

> Following flags may passed to :func:`compile` in order to change
> affects on compilation of a program:

Reviewer 2: ```suggestion
The following flags may be passed to :func:`compile` in order to change
effects on compilation of a program:
```

> The following flags may be passed to :func:`compile` in order to change
> effects on compilation of a program:

Reviewer: ```suggestion
The following flags may be passed to :func:`compile` in order to change
effects on the compilation of a program:
```

.. data:: PyCF_ALLOW_TOP_LEVEL_AWAIT

   Enables support for top-level ``await``, ``async for``, ``async with``
   and async comprehensions.

   .. versionadded:: 3.8

.. data:: PyCF_ONLY_AST

>    Generates and returns an abstract syntax tree instead of returning
>    compiled code object.

Reviewer: ```suggestion
   Generates and returns an abstract syntax tree instead of returning a
   compiled code object.
```
